ApplyToBecomeInternal/ApplyToBecomeInternal/Pages/TaskList/Decision/DeclineReason.cshtml.cs

```diff
@@ -0,0 +1,128 @@
+using ApplyToBecomeInternal.Services;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ApplyToBecomeInternal.Pages.TaskList.Decision
+{
+	public class DeclineReasonModel : DecisionBaseModel
+	{
+		private readonly ErrorService _errorService;
+
+		public DeclineReasonModel(ErrorService errorService, IAcademyConversionProjectRepository repository, ISession session) : base(repository, session)
+		{
+			_errorService = errorService;
+
+			DeclineOptions = Enum.GetValues(typeof(AdvisoryBoardDeclinedReasons)).Cast<AdvisoryBoardDeclinedReasons>();
+		}
+
+		[BindProperty, MinLength(length: 1, ErrorMessage = "ERROR MESSAGE TBC")]
+		public IEnumerable<string> DeclinedReasons { get; set; }
+
+		[BindProperty] public string DeclineOtherReason { get; set; }
+		[BindProperty] public string DeclineFinanceReason { get; set; }
+		[BindProperty] public string DeclinePerformanceReason { get; set; }
+		[BindProperty] public string DeclineGovernanceReason { get; set; }
+		[BindProperty] public string DeclineChoiceOfTrustReason { get; set; }
+
+		public IEnumerable<AdvisoryBoardDeclinedReasons> DeclineOptions { get; }
+
+		public UIHelpers UI => new UIHelpers(this);
+
+		public async Task<IActionResult> OnGetAsync(int id)
+		{
+			await SetDefaults(id);
+			PreloadStateFromSession(id);
+			SetBackLinkModel(Links.Decision.WhoDecided, id);
+
+			return Page();
+		}
+
+		private void PreloadStateFromSession(int id)
+		{
+			AdvisoryBoardDecision decision = GetDecisionFromSession(id);
+
+			DeclinedReasons = new StringValues(decision.DeclinedReasons?.Select(x => x.ToString()).ToArray());
+			DeclineOtherReason = decision.DeclinedOtherReason;
+			DeclineFinanceReason = decision.DeclineFinanceReason;
+			DeclinePerformanceReason = decision.DeclinePerformanceReason;
+			DeclineGovernanceReason = decision.DeclineGovernanceReason;
+			DeclineChoiceOfTrustReason = decision.DeclineChoiceOfTrustReason;
+		}
+
+		public async Task<IActionResult> OnPostAsync(int id, [FromQuery(Name = "obl")] bool overrideBackLink)
+		{
+			if (!ModelState.IsValid)
+			{
+				_errorService.AddErrors(ModelState.Keys, ModelState);
+				return await OnGetAsync(id);
+			}
+
+			var decision = GetDecisionFromSession(id);
+
+			decision.DeclinedReasons = DeclinedReasons.Select(Enum.Parse<AdvisoryBoardDeclinedReasons>).ToList();
+			decision.DeclinedOtherReason = DeclineOtherReason;
+			decision.DeclineFinanceReason = DeclineFinanceReason;
+			decision.DeclinePerformanceReason = DeclinePerformanceReason;
+			decision.DeclineGovernanceReason = DeclineGovernanceReason;
+			decision.DeclineChoiceOfTrustReason = DeclineChoiceOfTrustReason;
+
+			SetDecisionInSession(id, decision);
+
+			var destination = overrideBackLink ? Links.Decision.Summary.Page : Links.Decision.ApprovalDate.Page;
```

Comment:
This is no longer required. We discussed this after an issue @Lisa-V-Sau raised and for now on submit we will redirect to the next form in the wizard. There is no need to redirect back to the summary page.

Reply:
👍 implemented, thanks :)